Externals/mGBA/make_version.c.js

```diff
@@ -0,0 +1,134 @@
+var wshShell		= new ActiveXObject("WScript.Shell")
+var oFS				= new ActiveXObject("Scripting.FileSystemObject");
+
+wshShell.CurrentDirectory   += "\\mgba";
+var outfile                 = "../version.c";
+var cmd_commit              = " describe --always --abbrev=40 --dirty";
+var cmd_commit_short        = " describe --always --dirty";
+var cmd_branch              = " symbolic-ref --short HEAD";
+var cmd_rev                 = " rev-list HEAD --count";
+var cmd_tag                 = " describe --tag --exact-match";
+
+function GetGitExe()
+{
+	try
+	{
+		gitexe = wshShell.RegRead("HKCU\\Software\\GitExtensions\\gitcommand");
+		wshShell.Exec(gitexe);
+		return gitexe;
+	}
+	catch (e)
+	{}
+
+	for (var gitexe in {"git.cmd":1, "git":1, "git.bat":1})
+	{
+		try
+		{
+			wshShell.Exec(gitexe);
+			return gitexe;
+		}
+		catch (e)
+		{}
+	}
+
+	// last try - msysgit not in path (vs2015 default)
+	msyspath = "\\Git\\cmd\\git.exe";
+	gitexe = wshShell.ExpandEnvironmentStrings("%PROGRAMFILES(x86)%") + msyspath;
+	if (oFS.FileExists(gitexe)) {
+		return gitexe;
+	}
+	gitexe = wshShell.ExpandEnvironmentStrings("%PROGRAMFILES%") + msyspath;
+	if (oFS.FileExists(gitexe)) {
+		return gitexe;
+	}
+
+	WScript.Echo("Cannot find git or git.cmd, check your PATH:\n" +
+		wshShell.ExpandEnvironmentStrings("%PATH%"));
+	WScript.Quit(1);
+}
+
+function GetFirstStdOutLine(cmd)
+{
+	try
+	{
+		return wshShell.Exec(cmd).StdOut.ReadLine();
+	}
+	catch (e)
+	{
+		// catch "the system cannot find the file specified" error
+		WScript.Echo("Failed to exec " + cmd + " this should never happen");
+		WScript.Quit(1);
+	}
+}
+
+function GetFileContents(f)
+{
+	try
+	{
+		return oFS.OpenTextFile(f).ReadAll();
+	}
+	catch (e)
+	{
+		// file doesn't exist
+		return "";
+	}
+}
+
+// get info from git
+var gitexe = GetGitExe();
+var commit      	= GetFirstStdOutLine(gitexe + cmd_commit);
+var commit_short    = GetFirstStdOutLine(gitexe + cmd_commit_short);
+var branch  		= GetFirstStdOutLine(gitexe + cmd_branch);
+var rev	        	= GetFirstStdOutLine(gitexe + cmd_rev);
+var tag 	    	= GetFirstStdOutLine(gitexe + cmd_tag);
+var binary_name     = "mgba";
+var project_name    = "mGBA";
+var version_abi     = "0.9";
+var version_string  = "0.9.0";
```

Comment:
Yikes.

Reply:
This script reproduces the behavior of your version.cmake to make it consistent across platforms, this isn't even exposed to the user. May I know what the issue is here ?

Reply:
The issue is that this will almost certainly forget to get updated when the external is updated. If you don't pull it directly from the version.cmake file then it has an extremely large chance of getting out of sync. One source of truth is preferable.